src/nl802154.h

```diff
@@ -104,6 +106,22 @@ enum nl802154_attrs {
 
 	NL802154_ATTR_SUPPORTED_COMMANDS,
 
+    NL802154_ATTR_SCAN_TYPE,
```

Comment:
After just quickly throwing together some code, my first thoughts are about attribute naming. Looking at 802.15.4-2011, I can see how attributes for very simple MLME request / indication / confirm packets can be e.g. a u8 followed by another u8, with unique attribute names, but for slightly more complex attributes like the ED scan, it would probably be better to use a nested attribute.